motor/providers/awsec2ebs/setup.go

```diff
@@ -237,16 +241,36 @@ func CreateSnapshotFromVolume(ctx context.Context, cfg aws.Config, volID string,
 	*/
 
 	// wait for snapshot to be ready
+	time.Sleep(10 * time.Second)
 	snapProgress := *res.Progress
-	for !strings.Contains(snapProgress, "100") {
+	snapState := res.State
+	timeout := 0
+	for snapState != types.SnapshotStateCompleted || !strings.Contains(snapProgress, "100") {
 		log.Info().Str("progress", snapProgress).Msg("waiting for snapshot completion; sleeping 10 seconds")
 		time.Sleep(10 * time.Second)
 		snaps, err := ec2svc.DescribeSnapshots(ctx, &ec2.DescribeSnapshotsInput{SnapshotIds: []string{*res.SnapshotId}})
 		if err != nil {
+			var ae smithy.APIError
+			if errors.As(err, &ae) {
+				if ae.ErrorCode() == "InvalidSnapshot.NotFound" {
+					time.Sleep(30 * time.Second) // if it says it doesnt exist, even though we just created it, then it must still be busy creating
+					timeout++                    // still count against the timeout to avoid an eternal loop
```

Comment:
In theory, this can still end in an infinite loop because of the continue and the timeout check at the end of the loop. I'm not sure whether this could also happen in reality.

Reply:
true, good point, i increase the timeout but dont check it right here -- ill add a quick check just in case